editor/editor_vcs_interface.h

```diff
@@ -92,23 +138,39 @@ class EditorVCSInterface : public Object {
 	// Proxy functions to the editor for use
 	bool initialize(String p_project_root_path);
 	bool is_vcs_initialized();
-	Dictionary get_modified_files_data();
+	List<StatusFile> get_modified_files_data();
 	void stage_file(String p_file_path);
 	void unstage_file(String p_file_path);
 	void discard_file(String p_file_path);
 	void commit(String p_msg);
-	Array get_file_diff(String p_identifier, TreeArea p_area);
+	List<DiffFile> get_file_diff(String p_identifier, TreeArea p_area);
 	bool shut_down();
 	String get_project_name();
 	String get_vcs_name();
-	Array get_previous_commits();
-	Array get_branch_list();
+	List<Commit> get_previous_commits();
+	List<String> get_branch_list();
 	bool checkout_branch(String p_branch);
 	void pull();
 	void push();
 	void fetch();
 	void set_up_credentials(String p_username, String p_password);
-	Array get_line_diff(String p_file_path, String p_text);
+	List<DiffHunk> get_line_diff(String p_file_path, String p_text);
+
+	// Helper functions to create and convert Dictionary into data structures
+	Dictionary create_diff_line(int new_line_no, int old_line_no, String p_content, String p_status);
+	Dictionary create_diff_hunk(int old_start, int new_start, int old_lines, int new_lines);
+	Dictionary create_diff_file(String p_new_file, String p_old_file);
+	Dictionary create_commit(String p_msg, String p_author, String p_hex_id, int16_t p_time);
+	Dictionary create_status_file(String p_file_path, ChangeType p_change, TreeArea p_area);
+	Dictionary add_line_diffs_into_diff_hunk(Dictionary p_diff_hunk, Array p_line_diffs);
+	Dictionary add_diff_hunks_into_diff_file(Dictionary p_diff_file, Array p_diff_hunks);
+
+	DiffLine convert_diff_line(Dictionary p_diff_line);
+	DiffHunk convert_diff_hunk(Dictionary p_diff_hunk);
+	DiffFile convert_diff_file(Dictionary p_diff_file);
+	Commit convert_commit(Dictionary p_commit);
+	StatusFile convert_status_file(Dictionary p_status_file);
+
```

Comment:
All these helper functions can be private if I understand this correctly. I don't think they will need to be called from outside